src/imgix-core-js.js

```diff
@@ -43,15 +44,37 @@
         this.settings[key] = val;
       }
 
-      if (!Array.isArray(this.settings.domains)) {
+      // If settings.domains is passed in as an array of 1 or more elements
+      // will generate a deprecation warning
+      if (Array.isArray(this.settings.domains) && this.settings.domains.length > 0) {
```

Comment:
Don't we want to only raise this warning if `this.settings.domains.length > 1`? By definition, someone can't be using sharding if they're only providing one domain.

Reply:
That was more of a consequence of L24 where domains is defaulted to `[]`. But I agree with your point especially because I do the same thing a few more lines down.